simln-lib/src/lib.rs

```diff
@@ -313,6 +313,14 @@ pub trait LightningNode: Send {
 #[error("Destination generation error: {0}")]
 pub struct DestinationGenerationError(String);
 
+/// A trait for selecting destination nodes for payments in the Lightning Network.
+/// 
+/// This trait is responsible for choosing appropriate destination nodes from the existing network
+/// for payments to be sent to. The selection can be based on different strategies:
+/// - Predefined destinations (for deterministic payment patterns)
+/// - Random selection weighted by node capacity (for simulating network activity)
```

Comment:
I don't think we should have details about implementations in trait documentation, likewise below for `PaymentGenerator`